### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by NicolasT**
Some removals might be controversial...

### CODE REVIEW

**Comment by NicolasT** [comment]
Comments welcome, though I won't merge this yet, first enable & fix some more warnings-as-errors.

**Comment by NicolasT** [comment]
These changes no longer only 'remove dead code':
- Remove some dead code (according to coverage testing)
- Enable more warnings, most 'as error', and fix corresponding errors
- Re-instate coverage builds

File: src/node/statistics.ml [hunk 1]
```diff
@@ -241,18 +241,6 @@ module Statistics = struct
...
-    let x = new_op t start in
-    update_x_stats t.range_time_stats x
-
-  let new_range_entries t (start:float) =
-    let x = new_op t start in
-    update_x_stats t.range_time_stats x
-
-  let new_rev_range_entries t (start:float) =
-    let x = new_op t start in
-    update_x_stats t.rev_range_entries_time_stats x
```
Comment (domsj) [inline src/node/statistics.ml:254]: Hmm, don't these indicate that those statistics should still be captured? (Probably in sync_backend if I'm not mistaken)

File: src/node/catchup.ml [hunk 2]
```diff
@@ -84,21 +82,6 @@ let _with_client_connection ~tls_ctx (ips,port) f =
...
-    | None ->
-      if tlc_i = ( Sn.succ Sn.start )
-      then Lwt.return Store_1_behind
-      else Lwt.return Store_n_behind
-    | Some store_i when store_i = tlc_i -> Lwt.return Store_tlc_equal
-    | Some store_i when store_i > tlc_i -> Lwt.return Store_ahead
-    | Some store_i ->
-      if store_i = (Sn.pred tlc_i)
-      then Lwt.return Store_1_behind
-      else Lwt.return Store_n_behind
```
Comment (domsj) [inline src/node/catchup.ml:100]: The type `store_tlc_cmp` can now be removed too

**Comment by domsj** [comment]
Some minor comments but I definitely like cleaning up (y)

**Comment by NicolasT** [comment]
Reworking against 1.7.

File: src/node/statistics.ml (hunk 1 above)
Comment (NicolasT) [inline src/node/statistics.ml:254]: They are: there are other functions with the same names (but different types) defined below, so these ones weren't used at all.

File: src/node/catchup.ml (hunk 2 above)
Comment (NicolasT) [inline src/node/catchup.ml:100]: Will look into this in the forthcoming PR.

**Pull request closed without merging**